src/som/interpreter/LexicalScope.java

```diff
@@ -18,24 +18,54 @@
 
 public abstract class LexicalScope {
 
+  private final LexicalScope outer;
+
+  public LexicalScope(final LexicalScope outer) {
+    this.outer = outer;
+  }
+
+  public LexicalScope getContextualScope() {
+    return outer;
+  }
+
+  public MixinScope getOuterScope() {
```

Comment:
Same naming discussion as with the builders.
This needs to parallel them exactly.